cep-XXXX.md

```diff
@@ -0,0 +1,201 @@
+regex `^[a-z0-9](\.?[a-z0-9])*$`.
+
+The maximum length of a file extension MUST NOT exceed 16 characters.
+
+> The conda ecosystem currently recognizes two artifact extensions: `tar.bz2` and `conda`, versioned
+`v1` and `v2` respectively.
+
+#### Filenames
+
+Distributable conda artifacts MUST have a filename following this scheme:
+
+```text
+<package name>-<version string>-<build string>.<extension>
+```
+
+The maximum length of a filename MUST NOT exceed 192 characters.
+
+Virtual conda packages do not exist on disk and SHOULD NOT need filename standardization.
+
+#### Distribution strings
+
+A "distribution string" MAY be used to identify a package artifact without specifying the extension
+or the channel. It MUST match the following syntax:
+
+```text
+<package name>-<version string>-<build string>
+```
+
+Distribution strings apply to both distributable and virtual packages. They are used as the name of the
+directories where artifacts are extracted in the package cache, for example.
+
+> Note: Despite the similarity, distribution strings are not `MatchSpec`-like specifiers and
+> MUST NOT be used as such.
+
+### Identifying channels
+
+A conda channel is defined as a URL where one can find one or more `repodata.json` files arranged in one
+subdirectory (_subdir_) each. `noarch/repodata.json` MUST be present to consider the parent location a channel.
+
+#### Channel base URLs and names
+
+The base URL for the arbitrary location of a repodata file is defined as:
+
+```text
+<scheme>://[<authority>][/<path>/][/label/<label name>]/<subdir>/repodata.json
+```
+
+with `<scheme>`, `<authority>` and `<path>` defined by [RFC
+3986](https://datatracker.ietf.org/doc/html/rfc3986#section-3.2).
+
+Taken the channel definition above, the base URL without trailing slashes is thus:
+
+```text
+<scheme>://[<authority>][/<path>/][/label/<label name>]
+```
+
+For example, given `https://conda.anaconda.org/conda-forge/noarch/repodata.json`, the part leading
+to `noarch/repodata.json` and thus base URL is `https://conda.anaconda.org/conda-forge/conda-forge`. For
+local repodata such as `file:///home/username/channel/noarch/repodata.json`, the
+channel base URL is `file:///home/username/channel`.
+
+For convenience, the channel _name_ is defined as the concatenation of `scheme`, `authority` and
+`path` components. The `<scheme>://` part MAY be omitted if `scheme` is one of `http`, `https` or
+`file`. At least one of `authority` or `path` SHOULD be present. In their absence, the channel name
+MUST be considered empty, regardless the scheme. Empty channel names SHOULD NOT be used.
+
+When present, each path component MUST only contain lowercase ASCII letters, numbers, underscores,
+periods, and dashes. They MUST NOT start with a period or a dash. They SHOULD start and end with a
+letter or a number. If present, each path component MUST match this regex:
+
+```re
+^[a-z0-9_][a-z0-9_.-]*$
+```
```

Comment:
This might create a potential problem with `file://` URLs, especially when it comes to user home directories.

Reply:
I'll add an exception for `file://` channels, downgrading it to a SHOULD.